crypto/signature.re

```diff
@@ -1,27 +1,37 @@
 open Helpers;
 [@deriving (ord, eq)]
 type t =
-  | Ed25519(Ed25519.Signature.t);
+  | Ed25519(Ed25519.Signature.t)
+  | Secp256k1(Secp256k1.Signature.t);
 
 let sign = (secret, message) =>
   switch (secret) {
   | Secret.Ed25519(secret) => Ed25519(Ed25519.sign(secret, message))
+  | Secp256k1(secret) => Secp256k1(Secp256k1.sign(secret, message))
   };
 let verify = (key, signature, message) =>
   switch (key, signature) {
   | (Key.Ed25519(key), Ed25519(signature)) =>
     Ed25519.verify(key, signature, message)
+  | (Secp256k1(key), Secp256k1(signature)) =>
+    Secp256k1.verify(key, signature, message)
+  | (Ed25519(_) | Secp256k1(_), Ed25519(_) | Secp256k1(_)) => false
```

Comment:
```ocaml
| (Key.Secp256k1(_), Ed25519(_))
| (Key.Ed25519(_), Secp256k1(_)) => false
```

Reply:
Slightly different solution because when we add tz3 it will look different.